### PR TITLE
Add guide for github workflow to terraform azure resources

### PR DESCRIPTION
**Pull request by phalbert**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

## Added docs pages

Please also include the path for the added docs

- Deploy Azure Resource using Terraform (`/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md`)

### CODE REVIEW

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+# Deploy Azure Resource using Terraform
+
+In the following guide, you are going to create a self-service action in Port that executes a [GitHub workflow](/create-self-service-experiences/setup-backend/github-workflow/github-workflow.md) to deploy a [storage account](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview) in Azure using Terraform templates.
+
+
+## Example - creating a storage account
+
+Follow these steps to get started:
+
+1. Create the following GitHub action secrets:
```
Comment (PeyGis) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:16]: GitHub **Action**

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+   :::tip
+- `<GITHUB-ORG>` - your GitHub organization or user name.
+- `<GITHUB-REPO-NAME>` - your GitHub repository name.
+:::
+
+
+```json showLineNumbers
+[
+  {
+    "identifier": "github_terraform",
```
Comment (PeyGis) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:92]: You can make the identifier descriptive such as create_azure_storage
Same applies to the title of the Action blueprint

**Review by PeyGis — CHANGES_REQUESTED** [review]

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md (hunk 2 above)
Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:92]: Alright

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+In the following guide, you are going to create a self-service action in Port that executes a [GitHub workflow](/create-self-service-experiences/setup-backend/github-workflow/github-workflow.md) to deploy a [storage account](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview) in Azure using Terraform templates.
+
+
+## Example - creating a storage account
+
+Follow these steps to get started:
+
+1. Create the following GitHub Action secrets:
+    1. Create the Port credentials.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:17]:
```suggestion
    1. Create the following Port credentials:
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+        1. `PORT_CLIENT_ID` - Port Client ID [learn more](/build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
+        2. `PORT_CLIENT_SECRET` - Port Client Secret [learn more](/build-your-software-catalog/sync-data-to-catalog/api/#get-api-token).
+    2. Create the Azure Cloud credentials.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:20]:
```suggestion
    2. Create the following Azure Cloud credentials:
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+        :::tip
+        Follow this [guide](https://learn.microsoft.com/en-us/azure/developer/terraform/get-started-cloud-shell-bash?tabs=bash#create-a-service-principal) to create a service principal in order to get the Azure credentials.
+        :::
+        1. `ARM_CLIENT_ID` - Azure Client ID (APP ID) of the application.
+        2. `ARM_CLIENT_SECRET` - Azure Client Secret (Password) of the application.
+        3. `ARM_SUBSCRIPTION_ID` - Azure Subscription ID.
+        4. `ARM_TENANT_ID` - The Azure [Tenant ID](https://learn.microsoft.com/en-us/azure/azure-portal/get-subscription-tenant-id).
+
+2. Install Port's GitHub app by clicking [here](https://github.com/apps/getport-io/installations/new).
+
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:30]: The whole text is very crowded, please add a line of space between each article(1-6)

Comment (phalbert) [reply docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:30]: @hadar-co , do you mean `<br />`?

Comment (hadar-co) [reply docs/create-self-service-experiences/setup-backend/github-workflow/examples/create-azure-resource.md:30]: yes exactly

**Review by hadar-co — APPROVED** [review]

**Pull request merged**